packages/core/parcel-bundler/src/assets/KotlinAsset.js

```diff
@@ -0,0 +1,65 @@
+const Asset = require('../Asset');
+const localRequire = require('../utils/localRequire');
+
+class KotlinAsset extends Asset {
+  constructor(name, options) {
+    super(name, options);
+    this.type = 'js';
+  }
+
+  async generate() {
+    // require kotlin
+    const kotlinCompiler = await localRequire(
+      '@jetbrains/kotlinc-js-api',
+      this.name
+    );
+    const fs = require('fs', this.name);
+
+    // remove extension and path
+    let slashyIndex = this.id.lastIndexOf('/'); // linux/mac
+    if (slashyIndex === -1) slashyIndex = this.id.lastIndexOf('\\'); // windows
+    const fileName = this.id.substring(slashyIndex, this.id.lastIndexOf('.'));
```

Comment:
Can you use the `path` library for this? It will take care of the normalization cross platform for you.